src/interpreter.c

```diff
@@ -20,8 +20,98 @@ typedef struct {
     jl_value_t **locals; // slots for holding local slots and ssavalues
     jl_svec_t *sparam_vals; // method static parameters, if eval-ing a method body
     int preevaluation; // use special rules for pre-evaluating expressions
+    size_t ip; // Points to the currently-evaluating statement
 } interpreter_state;
 
+// Backtrace support;
+#ifdef _OS_LINUX_
+extern uintptr_t __start_jl_enter_interpreter_frame;
+extern uintptr_t __stop_jl_enter_interpreter_frame;
+
+extern uintptr_t __start_jl_interpreter_frame;
+extern uintptr_t __stop_jl_interpreter_frame;
+
+#define SECT_INTERP JL_SECTION("jl_interpreter_frame")
+#define SECT_ENTER_INTERP JL_SECTION("jl_enter_interpreter_frame")
+
+#elif defined(_OS_DARWIN_)
+extern uintptr_t __start_jl_enter_interpreter_frame_val __asm("section$start$__TEXT$__jeif");
+uintptr_t __start_jl_enter_interpreter_frame = (uintptr_t)&__start_jl_enter_interpreter_frame_val;
+extern uintptr_t __stop_jl_enter_interpreter_frame_val __asm("section$end$__TEXT$__jeif");
+uintptr_t __stop_jl_enter_interpreter_frame = (uintptr_t)&__stop_jl_enter_interpreter_frame_val;
+
+extern uintptr_t __start_jl_interpreter_frame_val __asm("section$start$__TEXT$__jif");
+uintptr_t __start_jl_interpreter_frame = (uintptr_t)&__start_jl_interpreter_frame_val;
+extern uintptr_t __stop_jl_interpreter_frame_val __asm("section$end$__TEXT$__jif");
+uintptr_t __stop_jl_interpreter_frame = (uintptr_t)&__stop_jl_interpreter_frame_val;
+
+#define SECT_INTERP JL_SECTION("__TEXT,__jif")
+#define SECT_ENTER_INTERP JL_SECTION("__TEXT,__jeif")
+#else
+JL_DLLEXPORT int jl_is_interpreter_frame(uintptr_t ip)
```

Comment:
Is this function used anywhere? Is this used for removing C frames in the interpreter?

Reply:
That's the idea, yes. Not implemented yet, though.

Reply:
Don't we do that already for all functions in `libjulia`?

Reply:
There's a mode where we display julia and C frames. I wanted to hide the interpreter frames in those traces, since we insert the reconstructed julia frame.